pkg/lib/docker/docker.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	dockertypes "github.com/docker/docker/api/types"
+	dockerclient "github.com/docker/docker/client"
+)
+
+var (
+	NoAuth string
+)
+
+func init() {
+	NoAuth, _ = EncodeAuthConfig(dockertypes.AuthConfig{})
+}
+
+func EncodeAuthConfig(authConfig dockertypes.AuthConfig) (string, error) {
+	encoded, err := json.Marshal(authConfig)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to encode docker login credentials")
+	}
+	registryAuth := base64.URLEncoding.EncodeToString(encoded)
+	return registryAuth, nil
+}
+
+func CheckImageAccessible(c *dockerclient.Client, dockerImage, registryAuth string) error {
+	_, err := c.DistributionInspect(context.Background(), dockerImage, registryAuth)
+	return err
```

Comment:
I wouldn't do this because then, you wouldn't get the full picture of why the operator couldn't pull the images - like in the errors I have referenced above this. I've double-checked, and the docker client errors come from `c.DistributionInspect`. Here's an example of an error that shouldn't be muted:
```
Error response from daemon: denied: User: arn:aws:iam::XXXXXXXXXXXX:user/cloud-user is not authorized to perform: ecr:BatchGetImage on resource: arn:aws:ecr:eu-central-1:XXXXXXXXXXXX:repository/mock-repository/hello-world with an explicit deny
```

What do you say?

Reply:
Yes, I am not proposing to hide any error messages, just to wrap, so maybe in this case it would be:

```
unable to access docker image XXXXX.dkr.ecr.us-west-2.amazonaws.com/XXXXX/XXXXX:latest: Error response from daemon: denied: User: arn:aws:iam::XXXXXXXXXXXX:user/cloud-user is not authorized to perform: ecr:BatchGetImage on resource: arn:aws:ecr:eu-central-1:XXXXXXXXXXXX:repository/mock-repository/hello-world with an explicit deny
```

given that, what do you think?

Reply:
Actually I see that this message is already incorporated in `ErrorImageInaccessible()`, correct?

Reply:
Oh yeah, definitely. They are already incorporated. Should I resolve this discussion?